app/src/main/java/org/sopt/sample/presentation/common/ViewModelFactory.kt

```diff
@@ -0,0 +1,27 @@
+package org.sopt.sample.presentation.common
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import org.sopt.sample.presentation.home.viewmodel.HomeViewModel
+import org.sopt.sample.presentation.signin.viewmodel.SignInViewModel
+import org.sopt.sample.presentation.signup.viewmodel.SignUpViewModel
+
+class ViewModelFactory : ViewModelProvider.Factory {
+
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        return when {
+            modelClass.isAssignableFrom(SignInViewModel::class.java) -> {
+                SignInViewModel() as T
+            }
+            modelClass.isAssignableFrom(SignUpViewModel::class.java) -> {
+                SignUpViewModel() as T
+            }
+            modelClass.isAssignableFrom(HomeViewModel::class.java) -> {
+                HomeViewModel() as T
+            }
+            else -> {
+                throw java.lang.IllegalArgumentException("Failed to create ViewModel: ${modelClass.name}")
```

Comment:
```suggestion
                throw IllegalArgumentException("Failed to create ViewModel: ${modelClass.name}")
```

Kotlin Exception을 활용해주세요!